### PR TITLE
[c#]Cycle Detection in a graph 

### LINKED ISSUE #4644: [c#]Cycle Detection in a graph
#### Proposed article introduction
Graphs are used to represent networks. The networks could include paths in a city or telephone network or circuit network. Graphs are also used in social networks like LinkedIn, Facebook. For example, in Facebook, each person is represented with a vertex(or node).
Cyclic graphs have several application in real life scenarios and it is therefore important that we get to understand how to detect it.

#### Key takeaways 
- What is a cycle in a graph.
-  How to detect a cycle using different algorithms giving an advantage of each. 
       1) Depth first search algorithm.
             - Cycle Detection on a directed graph
             - Cycle Detection on a undirected graph
- Understand different applications of cycle detection.

#### Article quality
This article will give a step by step process using well understood and simple visuals on how one can use the different algorithms to detect whether a graph has a cycle. 

#### References
https://www.section.io/engineering-education/graphs-in-data-structure-using-cplusplus/

### PR DESCRIPTION
**Pull request by erixbiz**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes  #4644

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by erixbiz** [comment]
@jumamiller  kindly have a look at this new PR. Gone through the article and others as well. I am optimistic that there is a flow now. Thank you.

**Comment by hectorkambow** [comment]
@jumamiller 

Just wanted to get a status on this PR. 

Lets be sure to NOT be assigning ourselves too many PRs at once - as we do not want our review time to be affected.

**Comment by hectorkambow** [comment]
cc @section-engineering-education/content-moderator-team

File: content/articles/graph-cycle-detection-c#/index.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
+A graph is a non-linear data structure that consists of:-
+
+1. Some grouped vertices are also called nodes.
+
+2. Some ordered pairs of the form (u, v) are called edges.
+
+
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:7]: Remove these extra spacings throughout your article!

File: content/articles/graph-cycle-detection-c#/index.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
+A graph is a non-linear data structure that consists of:-
+
+1. Some grouped vertices are also called nodes.
+
+2. Some ordered pairs of the form (u, v) are called edges.
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:5]: Could we have a better intro?

File: content/articles/graph-cycle-detection-c#/index.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+
+
+Graphs are one of the most versatile data structures. This is because they allow us to solve interesting problems. They are used in social networks and GPS. One can apply it anywhere you want to model the relationship between a bunch of objects. In this article, our primary focus will be on how to detect a cycle. It is important to know this concept to help us detect infinite loops in a computer program.
+
+
+#### Prerequisite
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:11]: Prerequisites?

File: content/articles/graph-cycle-detection-c#/index.md [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+
+To be able to follow this article well, one needs:
+
+1. To have [Visual Studio](https://visualstudio.microsoft.com/) installed.
+
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:16]: Kindly work on these spacings!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, kindly work on the intro part as suggested. Next work on your spacings where applicable!

**Comment by erixbiz** [comment]
@jumamiller  kindly have a look. Thank you

**Comment by hectorkambow** [comment]
@jumamiller 

can we get a status on this PR ?

If any more major revisions are needed - we will be closing this PR asap. 👀

@erixbiz 
Please be sure you are going over the entire article as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility.

**Comment by jumamiller** [comment]
Hi @erixbiz, kindly reach me on Slack ASAP!

**Comment by jumamiller** [comment]
Hi @erixbiz, kindly fix all image paths  to this:` graph-cycle-detection-c#` the current one is misses the `-c#` part...

**Review by jumamiller — APPROVED** [review]
Approved!

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@erixbiz 

This article is somewhat difficult to understand. 

There seems to be steps missing??

**Comment by hectorkambow** [comment]
@erixbiz 
How are we going from step 7 back to a number 6?? 

image

cc @jumamiller

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic suggestion. Not a problem but it seems that this PR is still quite a bit aways from being completed. 
 
In an effort to be fair to our larger EngEd community as a rule of thumb - we can only accept content (pull requests) that are complete and fully polished articles/tutorial - meaning they are ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) - this allow all revisions to be done on your forked personal repo. 
 
We will be closing this PR and please re-submit it when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 
 
>NOTE: A duplicate PR will go back into the queue and will not receive any further major revisions. A duplicate PR should be 100% ready to be published, otherwise it will be closed immediately. We encourage our new contributors to go over our [resources page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) and [new contributor folder](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) to be better familiar with the level of quality and style we are looking to publish.
 
Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
👍

**Pull request closed without merging**